python_modules/libraries/dagster-databricks/dagster_databricks/components/databricks_asset_bundle/component.py

```diff
@@ -69,6 +69,7 @@ def get_asset_spec(self, task: DatabricksBaseTask) -> AssetSpec:
                 "task_config": MetadataValue.json(task.task_config_metadata),
                 **({"libraries": MetadataValue.json(task.libraries)} if task.libraries else {}),
             },
+            deps=[snake_case(dep_task_key) for dep_task_key in task.depends_on],
```

Comment:
I think this makes it hard to work with if the user customizes the asset key. We should probably `get_asset_spec()` for the whole upstream task and then grab the key from it (like what we do for dbt). that way if the user customizes how the key is generated they don't have to also customize how the deps are generated